libs/SalesforceSDKCore/SalesforceSDKCore/Classes/RestAPI/SFRestAPI.m

```diff
@@ -196,12 +201,22 @@ - (void)send:(SFRestRequest *)request delegate:(id<SFRestDelegate>)delegate shou
             [[SFAuthenticationManager sharedManager] logout];
         }];
     } else {
-
-        // Auth credentials exist. Just send the request.
         [self enqueueRequest:request delegate:delegate shouldRetry:shouldRetry];
     }
 }
 
+- (SFOAuthSessionRefresher *)sessionRefresherForUser:(SFUserAccount *)user {
+    @synchronized (self) {
+
```

Comment:
You still need some kind of guard here.  In this case, it could be `self.oauthSessionRefresher == nil`.  But you don't want this value to be overwritten by multiple accesses, and that's not guarded against in the current code.